wgpu/src/util/belt.rs

```diff
@@ -99,10 +77,57 @@ impl StagingBelt {
         size: BufferSize,
         device: &Device,
     ) -> BufferViewMut<'_> {
+        let (mapped, belt_buffer, offset_in_belt_buffer) = self.allocate(
+            size,
+            const { BufferSize::new(crate::COPY_BUFFER_ALIGNMENT).unwrap() },
+            device,
+        );
+        encoder.copy_buffer_to_buffer(
+            belt_buffer,
+            offset_in_belt_buffer,
+            target,
+            offset,
+            size.get(),
+        );
+        mapped
+    }
+
+    /// Allocate a staging belt slice with the given `size` and `alignment` and return it.
+    ///
+    /// This allows you to do whatever you want with the slice after, such as
+    /// copying it to a texture or executing a compute shader that reads it, whereas
+    /// [`StagingBelt::write_buffer()`] can only write to other buffers.
+    ///
+    /// If the `size` is greater than the size of any free internal buffer, a new buffer
+    /// will be allocated for it. Therefore, the `chunk_size` passed to [`StagingBelt::new()`]
+    /// should ideally be larger than every such size.
+    ///
+    /// Three values are returned:
+    ///
+    /// * The mapped buffer view which you should write into from the CPU side (Rust code).
+    /// * The buffer containing the slice, for you to use in GPU commands.
+    ///   All commands involving this slice must be submitted after
+    ///   [`StagingBelt::finish()`] is called and before [`StagingBelt::recall()`] is called.
+    /// * The offset within the buffer at which the slice starts.
+    ///   This offset should be used for all GPU commands, and should not be used with
+    ///   the mapped buffer view.
+    pub fn allocate(
+        &mut self,
+        size: BufferSize,
+        alignment: BufferSize,
```

Comment:
What does this alignment represent? It appears to be the alignment of values *within* the chunk, right? Does this guarantee anything about the alignment of the returned `BufferViewMut`?

This should be documented somewhere.

Reply:
Added a paragraph on the alignment.